### ISSUE TITLE
Rules.md

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#137]
Thinking about documentation for 0.1.0.

Should we have a rules.md that lists (and importantly groups like #1) the available rules? Something like eslint does:

---

# General

* `indentation` - Specify the whitespace indentation.
* `no-eof-missing-newline` - Disallow missing newline at the end of a file
* `no-multiple-empty-lines` - Disallow multiple empty lines
* `no-eol-whitespace` - Disallow whitespace at the end of a line

# Number

* [`number-leading-zero`](https://github.com/stylelint/stylelint/blob/master/src/rules/number-leading-zero/README.md) - Require or disallow a leading zero for fractional value less than 1.
* `number-no-trailing-zeros` - Disallow trailing zeros

---

etc...

Would having a list like this also be helpful if the website is auto generated from the markdown?

**Comment by MoOx** [comment]
The list can be generated.

I will handle that really soon, engine is ready, cssnext website is also almost ready, so I will push the engine in here :)

**Comment by jeddy3** [comment]
> The list can be generated.

I was hoping that'd be the case :)

Will it be able to extract the rule name *and* the single line description from each of the markdown files? Being able to group the rules would be nice, but that's not essential for now.

> I will handle that really soon, engine is ready, cssnext website is also almost ready, so I will push the engine in here :)

Fantastic! I can't wait to see to see the new cssnext website and to take a look at the engine behind it  :)

**Comment by MoOx** [comment]
> Will it be able to extract the rule name and the single line description from each of the markdown files? 
rule name won't be a problem, for the desc I will see :)
Maybe we should add a header to md files by it won't be cool to get those in from of all README.md
I will try to handle that from a parsing.
For regrouping, we might need to define a property somewhere, maybe in the meta header 

```md
---
name: rule-name
desc: desc
category: category
```

actual content
```

**Comment by jeddy3** [comment]
> rule name won't be a problem, for the desc I will see :)

Sweet.

An alphabetically generated list of rule names that link through to the individual rule pages would be a great start and, I think, enough at the moment :)
